gnucash/gnome/gnc-budget-view.c

```diff
@@ -884,7 +884,7 @@ gbv_treeview_resized_cb (GtkWidget *widget, GtkAllocation *allocation,
         if ((g_strcmp0 (name, "description") == 0) && (!priv->show_account_desc))
             total_node = g_list_next (total_node);
 
-        if (gtk_tree_view_column_get_visible (tree_view_col))
+        if (gtk_tree_view_column_get_visible (tree_view_col) && total_node != NULL)
```

Comment:
Wouldn't it make more sense to just break out of the loop when you run out of totals columns?

OTOH the premise here seems to be that if show_account_code or show_account_desc is false then `gtk_tree_view_column_get_visible()` will return false for those columns anyway so you could just track through both in the `for` statement:
```
for (GList *node = columns, *total_node = total_columns;
       node && total_node;
       node = g_list_next (node), total_node = g_list_next (total_node))
{
     GtkTreeViewColumn *tree_view_col = node->data;
     if (gtk_tree_view_column_get_visible (tree_view_col))
     {
         gint col_width = gtk_tree_view_column_get_width (tree_view_col);
         GtkTreeViewColumn *totals_view_col = total_node->data;
         if (GTK_IS_TREE_VIEW_COLUMN(totals_view_col))
             gtk_tree_view_column_set_fixed_width (totals_view_col, col_width);
     }
}
```

But if the goal is to hide the `totals_view_col` for the not-displayed `tree_view_col` shouldn't it be
```
     GtkTreeViewColumn *tree_view_col = node->data;
     GtkTreeViewColumn *totals_view_col = total_node->data;
     gint col_width = 0;
     if (gtk_tree_view_column_get_visible (tree_view_col))
         col_width = gtk_tree_view_column_get_width (tree_view_col);
     if (GTK_IS_TREE_VIEW_COLUMN(totals_view_col))
         gtk_tree_view_column_set_fixed_width (totals_view_col, col_width);
```

instead?

Reply:
I'm not too familiar with the mechanics of gtk_tree_view_column; in this file columns and total_columns are of different lengths. In my budget with 2 periods, columns have 31 nodes (being derived from `gnc_tree_view_account_new` and has all acct properties) whereas total-columns have 6 nodes (presumably: acctname, desc, note, period1, period2, total) (?).

Reply:
Well, that explains why it over-runs. Maybe this function is a left over from a previous design? What it's trying to do is set the column widths of the totals_view to be the same as the corresponding tree_view ones, presumably so that they'll line up. Do the first n tree_view columns correspond to the n totals_view ones so that that would make sense?

Reply:
@christopherlam OK, I think. The last two columns entries correspond to totals and account currency?

Reply:
Uh no idea... Hence the patch to block crasher. Orig discussion https://lists.gnucash.org/pipermail/gnucash-devel/2013-February/034938.html

Reply:
Sigh, another half-baked feature creating long-term technical debt.

There's not much point in optimizing the loop, it's short and executing in user time. Go ahead and push.